sympy/polys/factortools.py

```diff
@@ -124,13 +124,29 @@ def dmp_trial_division(f, factors, u, K):
 
 
 def dup_zz_mignotte_bound(f, K):
-    """Mignotte bound for univariate polynomials in `K[x]`. """
-    a = dup_max_norm(f, K)
-    b = abs(dup_LC(f, K))
-    n = dup_degree(f)
-
-    return K.sqrt(K(n + 1))*2**n*a*b
+    """
+    The Knuth-Cohen variant of Mignotte bound for
+    univariate polynomials in `K[x]`.
 
+    Reference: John Abbott, Journal of Symbolic Computation 50 (2013) 532–563.
+    """
+    d = dup_degree(f)
+    delta = _ceil( d / 2 )
+    
+    # euclidean-norm
+    eucl_norm = K.sqrt( sum( [cf**2 for cf in f] ) )
+
+    # biggest values of binomial coefficients (p. 538 of reference)
+    t1 = binomial( delta - 1, _ceil( delta / 2 ) )
```

Comment:
`binomial` is not defined.

Reply:
You are right, checking the log now i see: "ImportError: cannot import name 'binomial' from partially initialized module 'sympy' (most likely due to a circular import)"
What is the correct way of importing the binomial function?

Reply:
If you want to resolve the circular import issue, you should import inside the function block itself.